site/app/controllers/UserProfileController.php

```diff
@@ -143,11 +144,13 @@ public function changeProfilePhoto() {
             return JsonResponse::getErrorResponse('No image uploaded to update the profile photo');
         }
         else {
-            $meta = explode('.', $_FILES['user_image']['name']);
-            $extension = $meta[1];
+            preg_match("/^.*\.(jpg|jpeg|png|gif)$/i", $_FILES['user_image']['name'], $extension);
+            if (!(FileUtils::isValidImage($_FILES['user_image']['tmp_name']) && FileUtils::validateUploadedFiles($_FILES['user_image'])[0]['success'] && (count($extension) >= 2) && $_FILES['user_image']['size'] <= 5 * 1048576)) {
+                return JsonResponse::getErrorResponse("Something's wrong with the uploaded file.");
+            }
 
```

Comment:
If I have a file such as image.png.file for example, it is considered valid by this. Is the intention of this to be valid?